google/src/main/scala/org/broadinstitute/dsde/rawls/google/GoogleUtilities.scala

```diff
@@ -21,31 +22,45 @@ import scala.util.{Failure, Success, Try}
 trait GoogleUtilities extends LazyLogging with InstrumentedRetry with GoogleInstrumented {
   implicit val executionContext: ExecutionContext
 
-  protected def when500orGoogleError(throwable: Throwable): Boolean = {
+  protected def when500or400orGoogleError(throwable: Throwable): Boolean = {
     throwable match {
       case t: HttpResponseException => t.getStatusCode/100 == 5 || t.getStatusCode/100 == 4
       case ioe: IOException => true
       case _ => false
     }
   }
 
-  // Sadly when500orGoogleError retries on 404s which is not always the behaviour we want. 404s
+  protected def when500orGoogleError(throwable: Throwable): Boolean = {
+    throwable match {
+      case t: HttpResponseException => t.getStatusCode/100 == 5
+      case ioe: IOException => true
+      case _ => false
+    }
+  }
+
+  // Sadly when500or400orGoogleError retries on 404s which is not always the behaviour we want. 404s
   // mean different things in different contexts. Sometimes they mean the resource does not
   // exist/you can't access it; sometimes it means it does not exist *yet* and sometimes something
   // else. As such, it should have been handled explicitly depending on the context and not rolled
   // into a default retrying handler. This ship has sailed - it's used everywhere and changing it
   // might cause a bunch of failures.
   def when500orNon404GoogleError(throwable: Throwable): Boolean = throwable match {
     case e: HttpResponseException if e.getStatusCode == 404 => false
-    case t => when500orGoogleError(t)
+    case t => when500or400orGoogleError(t)
+  }
+
+  //Predicate to exclude certain Google status codes from retries
+  def whenGoogleStatusDoesntContain(throwable: Throwable, excludedStatusCodes: Set[StatusCode]): Boolean = throwable match {
```

Comment:
Maybe you could curry this to avoid the call site looking grim: 
```scala 
def whenGoogleStatusDoesntContain(excludedStatusCodes: Set[StatusCode])(t: Throwable): Boolean
```

Then your call site is 
```scala 
retryExponentially(whenGoogleStatusCodeDoesntContain( ... )) { ... }
```